executor/src/witgen/machines/block_machine.rs

```diff
@@ -409,23 +458,27 @@ impl<T: FieldElement> BlockMachine<T> {
         next: bool,
         constraint: Constraint<T>,
         change_logger: &mut ChangeLogger<T>,
-    ) {
+    ) -> bool {
         let r = (self.row + next as DegreeType) % self.degree;
         match constraint {
             Constraint::Assignment(a) => {
                 change_logger.log(self, &name, &a, Some(r));
+                let r = r as usize;
                 let values = self.data.get_mut(&poly).unwrap();
-                if (r as usize) < values.len() {
+                if r < values.len() && values[r].is_none() {
```

Comment:
Shouldn't it be a hard error if it is some?

Reply:
No, because it might already have a value from the previous pass over the identities. Maybe I should add an assertion that the values should be the same in that case?

Reply:
Actually, if I fail hard in this case, things just work.